internal/metrics/listener.go

```diff
@@ -148,16 +150,51 @@ func getEnhancedMetricsTags(ctx context.Context) []string {
 	isColdStart := ctx.Value("cold_start")
 
 	if lc, ok := lambdacontext.FromContext(ctx); ok {
-		// ex: arn:aws:lambda:us-east-1:123497558138:function:golang-layer
+		// ex: aS
 		splitArn := strings.Split(lc.InvokedFunctionArn, ":")
+		var alias string
+		var executedversion string
+		var resource string
 
 		functionName := fmt.Sprintf("functionname:%s", lambdacontext.FunctionName)
 		region := fmt.Sprintf("region:%s", splitArn[3])
 		accountId := fmt.Sprintf("account_id:%s", splitArn[4])
 		memorySize := fmt.Sprintf("memorysize:%d", lambdacontext.MemoryLimitInMB)
 		coldStart := fmt.Sprintf("cold_start:%t", isColdStart.(bool))
-		return []string{functionName, region, accountId, memorySize, coldStart}
+
+		tags := []string{functionName, region, accountId, memorySize, coldStart}
+		if len(splitArn) > 7 {
+			alias = splitArn[7]
+		}
+		// Check we have an arn
+		if alias != "" {
+			// Drop the $ from $Latest based on ddog tagging convention
+			if strings.HasPrefix(alias, "$") {
+				alias = alias[1:]
+				// Check we have an alias and not a version. An alias can't be a number or start with $
+			} else if !isNumeric(alias) {
+				executedversion = fmt.Sprintf("executedversion:%s", lambdacontext.FunctionVersion)
+				tags = append(tags, executedversion)
+			}
+			resource = fmt.Sprintf("resource:%s:%s", lambdacontext.FunctionName, alias)
+		} else {
+			resource = fmt.Sprintf("resource:%s", lambdacontext.FunctionName)
```

Comment:
Since the `resource` tag will always be set, we can probably get rid of the `else` statement and the initialization of the `resource` variable on L157. So we can move this line to just before the `alias` conditional and change `=` to `:=`.